source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,166 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
+  {
+    AssertThrow (dim == 2,
+                 ExcMessage ("Only apply bound preserving limiter to the 2 dimensional problem"));
```

Comment:
Yes, right way. Let's leave it at 2d for now. Talk to me during the CIG workshop or the hackathon and I'll show you how to implement the general case.